### ISSUE TITLE
evolve throws an error

### ISSUE THREAD

**Issue by OrjwanZaafarani** [ultralytics/yolov5#3481]
## ❔Question
Why is this command throwing an error?
`python train.py --img 640 --batch 32 --epochs 100 --data QMUL.yaml --weights yolov5m6.pt --cache --evolve`

The error message:
```
Traceback (most recent call last):
  File "train.py", line 610, in <module>
    hyp[k] = max(hyp[k], v[1])  # lower limit
KeyError: 'anchors'
```

**Comment by OrjwanZaafarani** [comment]
I just solved the issue by following https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution#issuecomment-692599379

However, I still don't understand the effect of removing the anchors.

**Comment by glenn-jocher** [comment]
@OrjwanZaafarani for evolution you should uncomment the anchor field in your hyp file. This will allow the anchor count to evolve.

**Comment by glenn-jocher** [comment]
@OrjwanZaafarani good news 😃! Your original issue may now be fixed ✅ in PR #3887. This PR adds `hyp['anchors']=3` if the anchors key is missing/commented in your hyperparameter file, fixing a long-running problem when starting hyp evolution using the default command:
```
python train.py --evolve
```

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload with `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
